presto-main/src/main/java/com/facebook/presto/operator/Operator.java

```diff
@@ -73,6 +73,13 @@
     Page getOutput();
 
     /**
+     * After calling this method operator should revoke all reserved revocable memory
+     */
+    default void revokeMemory()
```

Comment:
I expect `revokeMemory` to work in background IO thread. There is no point in this method returning future completed when spilling has finished, since drivers/query will be blocked on out of memory future (which is released by freeing some memory). I expect background IO to call at some point `OperatorContext::freeRevocableMemory` or `OperatorContext::setRevocableMemoreReservation(0)`.